metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,24 +137,80 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         config = DeltaLakeSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    inner_data_type = ""
```

Comment:
f I understand correctly `self.inner_data_type` variable declared a source level captures the native type string for a complex column here.

Given that the delta-lake source may be used ingest multiple delta-tables with complex types Or same delta table with multiple fields having complex types, it is concerning that these variables are initialised globally once for source and reused across all tables without re-init. I suspect this needs change.

Reply:
fixed, resetting the value before calling the `_inner_field_details`

Reply:
does `counter` not need re-init ?

Also,  I'd prefer if the `_inner_field_details` is written such a way that it returns the parsed inner data type back in local scope of method `_parse_datatype`, instead of changing these class-level variables. Its harder to follow the code here.